sympy/matrices/dense.py

```diff
@@ -887,14 +887,14 @@ def hessian(f, varlist, constraints=()):
             raise ShapeError("`len(varlist)` must not be zero.")
     else:
         raise ValueError("Improper variable list in hessian function")
-    if not getattr(f, 'diff'):
+    if not f.diff:
```

Comment:
I'm guessing that this was supposed to be `hasattr`. Probably it should just be removed. Are there any tests covering this branch?

Reply:
No idea. If only we had coverage reports ;)

Reply:
It would be better to fix this somehow rather than just change it to something equally broken.